clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/provider/view/KubernetesJobProvider.java

```diff
@@ -73,14 +73,17 @@ public KubernetesJobStatus collectJob(String account, String location, String id
             jobStatus.getLocation(),
             KubernetesSelectorList.fromMatchLabels(selector));
 
-    jobStatus.setPods(
-        pods.stream()
-            .map(
-                p -> {
-                  V1Pod pod = KubernetesCacheDataConverter.getResource(p, V1Pod.class);
-                  return new KubernetesJobStatus.PodStatus(pod);
-                })
-            .collect(Collectors.toList()));
+    if (pods.size() <= credentials.getRunJobPodCollectionLimit()) {
```

Comment:
This is a bit confusing to me.

What is the use case for needing separate limits on the number of pods to collect, and the number of statuses to collect?   It feels like this is going to be complex for users to configure (adding to already complex configuration for kubernetes accounts).

Even with the default values, it's not clear to me what's happening.  If you have 5 pods, we'll collect them all with a status.  But if you have 6 pods, we won't collect any of them because we skip this block entirely.  (So what is `runJobPodStatusCollectionLimit = 10` controlling then?)

I think we can solve the problem with only a single configurable parameter controlling the number of jobs to collect (without the separate parameter for status), which would make it much easier for users to understand.